daos-cortx/src/daos_kv_benchmark/kv_benchmarking.cc

```diff
@@ -0,0 +1,448 @@
+   /** generate uuid for container */
+   uuid_generate( co_uuid );
+
+   /** create container */
+   rc = daos_cont_create( poh, co_uuid, NULL /* properties */,
+                          NULL /* event */ );
+   ASSERT( rc == 0, "container create failed with %d", rc );
+
+   /** open container */
+   rc = daos_cont_open( poh, co_uuid, DAOS_COO_RW, &coh, NULL,
+                        NULL );
+   ASSERT( rc == 0, "container open failed with %d", rc );
+
+   oid.hi = 0;
+   oid.lo = 4;
+
+   /** the KV API requires the flat feature flag be set in the oid */
+   daos_obj_generate_id( &oid, DAOS_OF_KV_FLAT, OC_SX, 0 );
+
+   rc = daos_kv_open( coh, oid, DAOS_OO_RW, &oh, NULL );
+   ASSERT( rc == 0, "KV open failed with %d", rc );
+}
+
+void tear_down( ) {
+   int rc;
+
+   /* close object handle */
+   daos_kv_close( oh, NULL );
+
+   /* close container */
+   rc = daos_cont_close( coh, NULL );
+   ASSERT( rc == 0, "cont close failed" );
+
+   /* delete container */
+   rc = daos_cont_destroy( poh, co_uuid, 1, NULL );
+   ASSERT( rc == 0, "cont delete failed" );
+
+   /* disconnect pool */
+   rc = daos_pool_disconnect( poh, NULL );
+   ASSERT( rc == 0, "disconnect failed" );
+
+   /* teardown the DAOS stack */
+   rc = daos_fini( );
+   ASSERT( rc == 0, "daos_fini failed with %d", rc );
+}
+
+/**
+ * @brief This function generates key name of key_size using number given number
+ *        Key format geneated by this function would be a string of characters
+ *        of key_size which starts with 'x' and last 16 bytes will be a number
+ *        'nr' with padding of zero before the number.
+ *
+ * @param key_buf    buffer to store key
+ * @param nr         key number used to generate unique key
+ * @param key_size   size of key
+ * @return           none
+ */
+void gen_key_name( char *key_buf, int nr, int key_size )
+{
+   char key_name[ 20 ] = {
+      0
+   };
+
+   /* generate different key */
+   memset( key_buf, 'x', key_size - 1 );
+   sprintf( key_name, "%.16d", nr );
+   strncpy( ( char * )key_buf + strlen( key_buf ) - strlen( key_name ), ( char * )key_name, strlen( key_name ) );
+}
+
+/* Benchmarking function to test KV Put operation */
+static void kv_put_function( benchmark::State &state ) {
+   int rc;
+
+   /* perform setup */
+   setup_main( );
+
+   unsigned int key_size = state.range( 0 ); // size of key
+   unsigned int val_size = state.range( 1 ); // size of value for the given key
+   unsigned int num_ops  = state.range( 2 ); // nr_ops is num of key-value pairs for the given test
+
+   /* allocate key and value buffers */
+   char *key_buf = ( char * )calloc( key_size, sizeof( char ) ); // key buffer allocated
+   char *val_buf = ( char * )calloc( val_size, sizeof( char ) ); // value buffer allocated
+   memset( val_buf, 'z', val_size - 1 );                         // populate with some random value.
+
+   /* actual computation starts here */
+   for ( auto _ : state )
+   {
+      /* call daos_kv_put for state.range(2) times */
```

Comment:
s/for state.range(2) times/for num_ops times

Reply:
please check same in other functions below

Reply:
done